src/classes/TG_ValidateData_TDTM.cls

```diff
@@ -0,0 +1,311 @@
+                    }
+                } else if (validator.needsContact()) {
+                    findContactList.add(termGrade);
+                } else if (validator.needsCourseOffering()) {
+                    findCourseOfferingList.add(termGrade); 
+                } else if (validator.needsContactAndCourseOffering()) {
+                    findContactCourseOfferingList.add(termGrade); 
+                } else if (validator.hasAllFields()) {
+                    // Course Connection, Contact and Course Offering are populated
+                    checkCourseContactList.add(termGrade);
+                }
+            }
+               
+            if (findCourseConnectionList != NULL && findCourseConnectionList.size() > 0) {           
+                findCourseConnection(findCourseConnectionList, termGradeContacts, termGradeCourseOffering, contactTermGradeMap);       
+            }
+               
+            if (checkCourseContactList != NULL && checkCourseContactList.size() > 0) {            
+                validateCourseContact(checkCourseContactList);        
+            }       
+               
+            if (findContactList != NULL && findContactList.size() > 0) {                               
+                findContact(findContactList);
+            }
+               
+            if (findCourseOfferingList != NULL && findCourseOfferingList.size() > 0) {               
+                findCourseOffering(findCourseOfferingList);     
+            }
+                
+            if (findContactCourseOfferingList != NULL && findContactCourseOfferingList.size() > 0) {           
+                findContactCourseOffering(findContactCourseOfferingList);
+            }       
+       }
+    
+        return dmlWrapper;
+    }
+    
+    /*******************************************************************************************************
+    * @description Queries for a Course Connection for the Term Grade based on the Contact and Course Offering.
+    * Updates the Term Grade if a Course Connection is found and throws an error if no Course Connection is found
+    * or more than 1 Course Connections are found.
+    * @param termGrades A List of Term Grades with no Course Connection.
+    ********************************************************************************************************/
+    
+    private void findCourseConnection(List<Term_Grade__c> termGrades, List<Id> termGradeContacts, List<Id> termGradeCourseOffering, Map<Id, List<Term_Grade__c>> contactTermGradeMap) {
+            
+        Map<Term_Grade__c, List<Course_Enrollment__c>> termGradeEnrollmentMap = new Map<Term_Grade__c, List<Course_Enrollment__c>>();
+        
+        for (Course_Enrollment__c eachEnrollment: [SELECT Id, Contact__c, Course_Offering__c FROM Course_Enrollment__c WHERE Contact__c IN:termGradeContacts AND Course_Offering__c IN:termGradeCourseOffering]) {        
```

Comment:
@skosgi1 -- missed the space here on the colon "eachEnrollment : [SELECT...." and then let's add some space after the IN, so IN :termGradeContacts (I'm ripping my style choices from here: https://developer.salesforce.com/docs/atlas.en-us.apexcode.meta/apexcode/langCon_apex_SOQL_variables.htm)

Reply:
(sorry I didn't catch those before, my bad)